### DISCUSSION TITLE
Documenting that the default IPv6 route set by router advertisment gets indirectly disabled by the Docker bridge

### DISCUSSION THREAD

**Discussion by EverSoJoe** [nextcloud/all-in-one#4400] [Questions] [answered]
### Is your feature request related to a problem? Please describe. 

Due to docker setting up the IPv6 forwarding kernel feature, the `accept_ra` (router advertisment) feature is being overwritten by default, as Linux assumes itself becoming the router when forwarding is enabled. This creates an issue of the default IPv6 route not being gotten by and set from router advertisment, meaning Linux does not know where to send any IPv6 traffic outside of the local network.
This can be circumvented by setting the `accept_ra` to 2, which means to accept router advertisment regardless of forwarding.

As far as I know, this issue only arises if IPv6 is **not** set up manually

### Describe the solution you'd like

The solution I came up with and that worked for me is to create a systemd unit, that runs after docker but before the networking that sets `accept_ra=2` on the default network interface. This should be mentioned in the documentation.

Example systemd unit file (interface name needs to be replaced):
```
[Unit]
DefaultDependencies=no
Description=accept_ra on eth0
Before=network.target
After=docker.service
Requires=docker.service

[Service]
Type=oneshot
ExecStart=/usr/sbin/sysctl -w net.ipv6.conf.eth0.accept_ra=2
TimeoutSec=0
RemainAfterExit=yes

[Install]
WantedBy=network.target
```

### Describe alternatives you've considered

I have only considered but never actually tested these solutions:
* static routing on the host
* not using the Docker bridge (as it seems to be a legacy feature anyway https://docs.docker.com/network/drivers/bridge/#use-the-default-bridge-network)

### Additional context

I only found this issue when I tried to set up Uptime Kuma monitoring for my Nextcloud instance and it prefers IPv6 whenever possible. Went down a debugging rabbithole from there until I found no default route on the `ip -6 route` output of the host.
Fixing it with the sysctl.conf didn't work as Docker seems to set this variable itself on startup. Which is why I resorted to the unit file to set it to the needed value after Docker was started.

**Comment by szaimen** [comment]
Hi, have you followed https://github.com/nextcloud/all-in-one/blob/main/docker-ipv6-support.md?

**Comment by EverSoJoe** [comment]
> Hi, have you followed https://github.com/nextcloud/all-in-one/blob/main/docker-ipv6-support.md?

Hi. Yes. I have followed this exactly. I installed on a Proxmox LXC container. Other container of mine don't have any issues with IPv6 connectivity and after my change it worked flawlessly as well

**Comment by szaimen** [comment, marked as answer]
> sysctl -w net.ipv6.conf.eth0.accept_ra=2

Honestly this sounds like something that needs to be fixed either in docker or the kernel that you are using. So please report it accordingly to upstrea. Thanks!
